src/Service/OutputCacheService.php

```diff
@@ -0,0 +1,137 @@
+use Psr\Container\ContainerInterface;
+use Pimcore\Logger;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class OutputCacheService 
+{
+    /**
+     * @var bool
+     */
+    private $cacheEnabled = false;
+    
+    /**
+     * The cached items lifetime in seconds
+     * 
+     * @var int
+     */
+    private $lifetime = 30;
+    
+    /**
+     * @var EventDispatcherInterface
+     */
+    public $eventDispatcher;
+
+    
+    /**
+     * @param ContainerInterface $container
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function __construct(ContainerInterface $container, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->eventDispatcher = $eventDispatcher;
+
+        $config = $container->getParameter('pimcore_data_hub');
+
+        if (isset($config['graphql'])) {
+            if (isset($config['graphql']['output_cache_enabled'])) {
+                $this->cacheEnabled = filter_var($config['graphql']['output_cache_enabled'], FILTER_VALIDATE_BOOLEAN);
+            }
+            
+            if (isset($config['graphql']['output_cache_lifetime'])) {
+                $this->lifetime = intval($config['graphql']['output_cache_lifetime']);
+            }
+
+            if (isset($config['graphql']['output_cache_exclude_pattern'])) {
+                $this->exclude_pattern = $config['graphql']['output_cache_exclude_pattern'];
+            }
+        }
+    }
+    
+    
+    public function load(Request $request) {
+        
+        if(!$this->useCache($request)) {
+            return null;
+        }
+        
+        $cacheKey = $this->computeKey($request);
+        
+        return $this->loadFromCache($cacheKey);
+    }
+    
+    
+    public function save(Request $request, JsonResponse $response, $extraTags = []) : void {
+        if ($this->useCache($request)) {
+            $cacheKey = $this->computeKey($request);
+            $clientname = $request->get('clientname');
+            $extraTags = array_merge(["output","datahub", $clientname], $extraTags);
+            
+            $event = new OutputCachePreSaveEvent($request, $response);
+            $this->eventDispatcher->dispatch(OutputCacheEvents::PRE_SAVE, $event);
+
+            $this->saveToCache($cacheKey, $event->getResponse(), $extraTags);
+        }
+    }
+    
+    protected function loadFromCache($key) {
+        return \Pimcore\Cache::load($key);
+    }
+    
+    protected function saveToCache($key, $item, $tags = []) : void { 
+        \Pimcore\Cache::save($item, $key, $tags, $this->lifetime);
+    }
+    
+    private function computeKey(Request $request) : string {
+        $clientname = $request->get('clientname');
+        
+        $input = json_decode($request->getContent(), true);
+        $input = print_r($input, true);
+        
+        return md5($clientname . $input);
```

Comment:
IMHO, we should prefix this with something (just as we do it for object_, asset_, etc)